### ISSUE TITLE
[mesa] Unable to build

### ISSUE THREAD

**Issue by uberhacker** [skycocker/chromebrew#4505]
```
In file included from ../src/util/u_math.h:42:
../include/c99_math.h:194:10: fatal error: 'cmath' file not found
#include <cmath>
         ^~~~~~~
1 error generated.
[94/1781] Compiling C++ object src/compiler/libcompiler.a.p/nir_types.cpp.o
In file included from ../src/compiler/nir_types.cpp:29:
In file included from ../src/compiler/glsl/ir.h:37:
../src/compiler/glsl/ir_hierarchical_visitor.h:77:7: warning: 'ir_hierarchical_visitor' has virtual functions but non-virtual destructor [-Wnon-virtual-dtor]
class ir_hierarchical_visitor {
      ^
1 warning generated.
[95/1781] Compiling C object src/mapi/es2api/libGLESv2.so.2.0.0.p/.._entry.c.o
[96/1781] Compiling C object src/mapi/shared-glapi/libglapi.so.0.0.0.p/.._entry.c.o
[97/1781] Compiling C object src/util/format/libmesa_format.a.p/meson-generated_.._u_format_table.c.o
[98/1781] Generating nir_opt_algebraic.c with a meson_exe.py custom command
ninja: build stopped: subcommand failed.
mesa failed to build: `ninja -C builddir` exited with 1
```

**Comment by uberhacker** [comment]
Also on x86_64:
```
[94/2305] Compiling C++ object src/compiler/libcompiler.a.p/glsl_types.cpp.o
FAILED: src/compiler/libcompiler.a.p/glsl_types.cpp.o 
clang -Isrc/compiler/libcompiler.a.p -Isrc/compiler -I../src/compiler -Isrc/mapi -I../src/mapi -Isrc/mesa -I../src/mesa -Iinclude -I../include -Isrc -I../src -I../src/gallium/include -Isrc/gallium/auxiliary -I../src/gallium/auxiliary -I/usr/local/include/valgrind -fvisibility=hidden -Xclang -fcolor-diagnostics -DNDEBUG -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++14 -O3 -Wl,--gc-sections -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS '-DPACKAGE_VERSION="20.2.1"' '-DPACKAGE_BUGREPORT="https://gitlab.freedesktop.org/mesa/mesa/-/issues"' -DUSE_ELF_TLS -DHAVE_ST_VDPAU -DENABLE_ST_OMX_BELLAGIO=0 -DENABLE_ST_OMX_TIZONIA=0 -DHAVE_X11_PLATFORM -DGLX_INDIRECT_RENDERING -DGLX_DIRECT_RENDERING -DGLX_USE_DRM -DHAVE_DRM_PLATFORM -DENABLE_SHADER_CACHE -DHAVE___BUILTIN_BSWAP32 -DHAVE___BUILTIN_BSWAP64 -DHAVE___BUILTIN_CLZ -DHAVE___BUILTIN_CLZLL -DHAVE___BUILTIN_CTZ -DHAVE___BUILTIN_EXPECT -DHAVE___BUILTIN_FFS -DHAVE___BUILTIN_FFSLL -DHAVE___BUILTIN_POPCOUNT -DHAVE___BUILTIN_POPCOUNTLL -DHAVE___BUILTIN_UNREACHABLE -DHAVE_FUNC_ATTRIBUTE_CONST -DHAVE_FUNC_ATTRIBUTE_FLATTEN -DHAVE_FUNC_ATTRIBUTE_MALLOC -DHAVE_FUNC_ATTRIBUTE_PURE -DHAVE_FUNC_ATTRIBUTE_UNUSED -DHAVE_FUNC_ATTRIBUTE_WARN_UNUSED_RESULT -DHAVE_FUNC_ATTRIBUTE_WEAK -DHAVE_FUNC_ATTRIBUTE_FORMAT -DHAVE_FUNC_ATTRIBUTE_PACKED -DHAVE_FUNC_ATTRIBUTE_RETURNS_NONNULL -DHAVE_FUNC_ATTRIBUTE_ALIAS -DHAVE_FUNC_ATTRIBUTE_NORETURN -DHAVE_FUNC_ATTRIBUTE_VISIBILITY -DHAVE_UINT128 -D_GNU_SOURCE -DUSE_SSE41 -DUSE_GCC_ATOMIC_BUILTINS -DUSE_X86_64_ASM -DMAJOR_IN_SYSMACROS -DHAVE_LINUX_FUTEX_H -DHAVE_ENDIAN_H -DHAVE_DLFCN_H -DHAVE_EXECINFO_H -DHAVE_SYS_SHM_H -DHAVE_STRTOF -DHAVE_MKOSTEMP -DHAVE_TIMESPEC_GET -DHAVE_MEMFD_CREATE -DHAVE_RANDOM_R -DHAVE_FLOCK -DHAVE_STRTOK_R -DHAVE_GETRANDOM -DHAVE_PROGRAM_INVOCATION_NAME -DHAVE_POSIX_MEMALIGN -DHAVE_DIRENT_D_TYPE -DHAVE_STRTOD_L -DHAVE_DLADDR -DHAVE_DL_ITERATE_PHDR -DHAVE_ZLIB -DHAVE_ZSTD -DHAVE_PTHREAD -DHAVE_PTHREAD_SETAFFINITY -DHAVE_LIBDRM -DLLVM_AVAILABLE '-DMESA_LLVM_VERSION_STRING="10.0.0"' -DLLVM_IS_SHARED=1 -DHAVE_VALGRIND -DHAVE_LIBUNWIND -DHAVE_WAYLAND_PLATFORM -DWL_HIDE_DEPRECATED -DHAVE_DRI3 -DHAVE_DRI3_MODIFIERS -Qunused-arguments -fuse-ld=lld -fPIC -MD -MQ src/compiler/libcompiler.a.p/glsl_types.cpp.o -MF src/compiler/libcompiler.a.p/glsl_types.cpp.o.d -o src/compiler/libcompiler.a.p/glsl_types.cpp.o -c ../src/compiler/glsl_types.cpp
In file included from ../src/compiler/glsl_types.cpp:25:
In file included from ../src/mesa/main/macros.h:35:
In file included from ../src/util/u_math.h:42:
../include/c99_math.h:194:10: fatal error: 'cmath' file not found
#include <cmath>
         ^~~~~~~
1 error generated.
[95/2305] Linking target src/mapi/es1api/libGLESv1_CM.so.1.1.0
[96/2305] Linking target src/mapi/es2api/libGLESv2.so.2.0.0
ninja: build stopped: subcommand failed.
mesa failed to build: `ninja -C builddir` exited with 1
```

**Comment by satmandu** [comment]
That last bit is a clang error I've seen, but not when gcc is building.

**Comment by uberhacker** [comment]
Fixed in #4550.
